neo4j-jdbc-bolt/src/main/java/org/neo4j/jdbc/bolt/BoltNeo4jResultSet.java

```diff
@@ -110,9 +113,20 @@ public class BoltNeo4jResultSet extends Neo4jResultSet {
 	// visible for testing
 	BoltNeo4jResultSet(Statement statement, Result iterator, int... params) {
 		super(statement, params);
-		List<Record> recordList = iterator != null ? iterator.list() : Collections.emptyList();
-		Optional<Record> first = recordList.stream().findFirst();
-		this.iterator = iterator != null ? recordList.iterator() : null;
+		Optional<Record> first = iterator != null && iterator.hasNext() ? Optional.of(iterator.peek()) : Optional.empty();
+		try {
+			// users that are not using autocommit would not materialize the result in memory
+			if (iterator != null
```

Comment:
this is OK for now, I will open an issue to handle autocommit properly in the near future

Reply:
https://github.com/neo4j-contrib/neo4j-jdbc/issues/377